### PR TITLE
feature 7 cart page

### PR DESCRIPTION
**Pull request by MarkianV**
Cart is complete and thoroughly documented. The errors are due to the fact Listing hasn't been implemented yet.

//TODO has been written in all places where work is not complete.

CartPage is incomplete but the logic is basically there. It's uncomfortable to finish because it heavily relies on the GUI and we haven't quite figured out how we will be using View and rendering pages. CartPage is also very roughly documented but for now, that should work. Any errors are due to the fact it is not complete and can be safely ignored for now.

### CODE REVIEW

**Review by danielqiuu — APPROVED** [review]
Good starting point for the GUI

**Review by claregillis — APPROVED** [review]
Looks good, nice documentation mark

File: src/main/java/features/Cart.java [hunk 1]
```diff
@@ -1,4 +1,88 @@
...
+     *
+     * @param itemListings an ArrayList of unique Listings.
+     */
+    public Cart(ArrayList<Listing> itemListings){
+        this.listings = itemListings;
+    }
+
+    //OVERLOADED
+    /** A Constructor for Cart that takes no arguments*/
+    public Cart(){
```
Comment (lukecheseldine) [inline src/main/java/features/Cart.java:20]: good thinking w/ two constructors

File: src/main/java/features/Cart.java [hunk 2]
```diff
@@ -1,4 +1,88 @@
...
+        this.listings = new ArrayList<Listing>();
+    }
+
+
+    /** Get all Listings in the cart.
+     *
+     * @return A shallow copy of the list of Listings within Cart (An ArrayList of Listings).
+     */
+    public ArrayList<Listing> getItems(){
```
Comment (lukecheseldine) [inline src/main/java/features/Cart.java:29]: going forward, i don't think it's necessary to create a shallow copy and you probs don't have to be so thorough. that said, this is really smart and much safer. good job :)

**Review by lukecheseldine — APPROVED** [review]
really really good mark!i have no real feedback, but there was some styling inconsistency. can you just reformat your code in intellij before you merge? great job :)

File: src/main/java/features/Cart.java (hunk 1 above)
Comment (helenastra) [reply src/main/java/features/Cart.java:19]: great documentation! makes it very clear to understand what is happening and where.

File: src/main/java/features/Cart.java (hunk 2 above)
Comment (MarkianV) [reply src/main/java/features/Cart.java:28]: Thanks for the feedback man, I appreciate it immensely. I've fixed any styling inconsistency that I saw, just now. If there are any others, I don't see them and would appreciate you pointing them out.

**Review by lukecheseldine — DISMISSED** [review]
great job!

**Review by danielqiuu — APPROVED** [review]

**Review by liu-samuel — APPROVED** [review]
lgtm

**Review by ethanrstark — APPROVED** [review]

**Review by helenastra — APPROVED** [review]

**Review by danielqiuu — APPROVED** [review]

**Pull request merged**